Require authentication for deleting room aliases

jooert opened the pull request:
This adds an authentication middleware to DELETE
/directory/room/:room_alias, as mandated by the spec, and updates the
test accordingly. The test now actually checks the status code of the
response to the DELETE request. Furthermore, the `delete` helper method
in `test.rs` has been modified to not require a body.

Marks the endpoint as done in STATUS.md.

jimmycuadra commented:
The code looks good. I think we should wait to mark the endpoint as ✅ , since it should also have an authorization check that the requesting user has some sort of minimum power level for the room being aliased. Since we don't support power levels yet, I'd say mark this endpoint as 🚧 .

jooert commented:
Done. i have also updated some comments in `directory.rs` and removed a `map_err` call in `room_alias.rs` because a DELETE will not return `DieselError::NotFound` and the spec doesn't mention that deleting a non-existing alias should be treated differently.
